x-pack/plugins/security_solution/public/onboarding/components/onboarding_body/cards/integrations/integrations_check_complete.ts

```diff
@@ -5,23 +5,63 @@
  * 2.0.
  */
 
+import type { GetPackagesResponse } from '@kbn/fleet-plugin/public';
+import { EPM_API_ROUTES, installationStatuses } from '@kbn/fleet-plugin/public';
+import { i18n } from '@kbn/i18n';
+import { lastValueFrom } from 'rxjs';
 import type { OnboardingCardCheckComplete } from '../../../../types';
-import { getDummyAdditionalBadge } from './integrations_header_badges';
+import { AGENT_INDEX } from './constants';
+import type { StartServices } from '../../../../../types';
 
-export const checkIntegrationsCardComplete: OnboardingCardCheckComplete = async () => {
-  // implement this function
-  return new Promise((resolve) =>
-    setTimeout(
-      () =>
-        resolve({
-          isComplete: true,
-          completeBadgeText: '3 integrations installed',
-          additionalBadges: [getDummyAdditionalBadge()],
-          metadata: {
-            integrationsInstalled: 3,
-          },
-        }),
-      2000
-    )
+export const checkIntegrationsCardComplete: OnboardingCardCheckComplete = async (
+  services: StartServices
+) => {
+  const packageData = await services.http.get<GetPackagesResponse>(
+    EPM_API_ROUTES.INSTALL_BY_UPLOAD_PATTERN,
+    {
+      version: '2023-10-31',
+    }
   );
+
+  const agentsData = await lastValueFrom(
+    services.data.search.search({
+      params: { index: AGENT_INDEX, body: { size: 1 } },
+    })
+  );
+
+  const installed = packageData?.items?.filter(
+    (pkg) =>
+      pkg.status === installationStatuses.Installed ||
+      pkg.status === installationStatuses.InstallFailed
```

Comment:
What does this package status mean? Should we count them as successfully installed?

Reply:
In the Fleet page, they regard these two status as installed: https://github.com/elastic/kibana/blob/7ea0ddc038337181df055f3439969706601d7103/x-pack/plugins/fleet/public/applications/integrations/sections/epm/screens/home/index.tsx#L63-L71

So I'm following the same implementation to have the consistent count